sync-core/src/main/java/com/cloudant/http/HttpConnection.java

```diff
@@ -140,17 +143,27 @@ public void setRequestBody(InputStream input, long inputLength) {
     public HttpConnection execute() throws IOException {
         System.setProperty("http.keepAlive", "false");
         connection = (HttpURLConnection) url.openConnection();
-        for (String key : requestProperties.keySet()) {
-            connection.setRequestProperty(key, requestProperties.get(key));
-        }
         // always read the result, so we can retrieve the HTTP response code
         connection.setDoInput(true);
         connection.setRequestMethod(requestMethod);
-        if (contentType != null) {
+
+        // Set request headers
+        List<String> lowerCaseExistingHeaders = new ArrayList<String>();
+        for (String key : requestProperties.keySet()) {
+            connection.setRequestProperty(key, requestProperties.get(key));
+            lowerCaseExistingHeaders.add(key.toLowerCase());
+        }
+
+
+        if (contentType != null && !lowerCaseExistingHeaders.contains("content-type")) {
             connection.setRequestProperty("Content-type", contentType);
         }
-        connection.setRequestProperty("User-Agent", userAgent);
-        if (url.getUserInfo() != null) {
+
+        if (!lowerCaseExistingHeaders.contains("user-agent")) {
+            connection.setRequestProperty("User-Agent", userAgent);
```

Comment:
We should probably deny the ability to set the user agent, then open if up for customization, rather than allowing it to be set to anything, then restricting it in the future.